lib/spack/spack/cmd/compiler.py

```diff
@@ -91,7 +104,48 @@ def compiler_find(args):
        add them to Spack's configuration.
 
     """
-    paths = args.add_paths
+    paths = args.add_paths or []
+
+    # Specs
+    if args.specs:
+        abstract_specs = spack.cmd.parse_specs(args.specs)
+        concretized = [item.concretized() for item in abstract_specs]
+        Compiler = collections.namedtuple(
+            'Compiler', 'request,abstract,concretized'
+        )
+        specs = [
+            Compiler(x, y, z) for x, y, z
+            in zip(args.specs, abstract_specs, concretized)
+        ]
+
+        # Check that the spec is that of a compiler
+        supported = spack.compilers.supported_compilers() + ['llvm']
+        not_compilers = [x for x in specs if x.abstract.name not in supported]
+        if not_compilers:
+            msg = 'these packages are not compilers [{0}]'
```

Comment:
maybe make that "spack-supported compilers"

Reply:
If we decide to continue with this PR, it's probably worth to mark in some way the packages that provide a compiler (a `compiler` tag?) and rely on that for the check above. Currently, for instance, `flang` and `intel-parallel-studio` are not accounted as specs that provide a compiler.

Reply:
The question is more what are we talking about long-term wise. Will compilers become "normal" build dependencies? Then I'd put here a "spack-supported" tag as this is really all it will amount to. If compilers will remain a thing, then we need to have a larger discussions on what constitutes a compiler that is recognised by spack (i.e. there are other languages besides C/C++ and Fortran that are compiled, do they get their own compiler support?)

I really don't have the knowledge to offer a well founded opinion, but it feels like wack a mole to keep up with all possible compilers. And (at least from my well of ignorance) there isn't much difference between `py-setuptools` and `gcc`, as in both modify the installed package and therefore (sometimes) need to be consistent within a runtime environment.